docs/current/guides/snippets/multi-builds/index.mjs

```diff
@@ -0,0 +1,46 @@
+import Client, { connect } from "@dagger.io/dagger"
+
+// Create a multi-build pipeline for a Go application.
+
+// define build matrix
+const oses = ["linux", "darwin"]
+const arches = ["amd64", "arm64"]
+
+// initialize dagger client
+connect(async (client: Client) => {
+    console.log("Building with Dagger")
+
+    // get reference to the local project
+    const src = await client
+        .host()
+        .directory(".")
+
+    // create empty directory to put build outputs
+    var outputs = client.directory()
+
+    const golang = await client.container()
+        // get golang image
+        .from("golang:latest")
+        // mount source code into golang image
+        .withMountedDirectory("/src", src)
+        .withWorkdir("/src")
+
+    await Promise.all(oses.map(async (os) => {
+        await Promise.all(arches.map(async (arch) => {
```

Comment:
You don't need `await Promise.all` here. Simple loop is fine. It's just adding directories to a lazy pipeline. Only `await` needed is on `.export`.

Reply:
done

Reply:
Updated for review